### PR TITLE
add blacklist function

### PR DESCRIPTION
**Pull request by geb789**
What does this PR do?

1. Add an account into black list;
2. Delete an account from black list;
3. Check  whether an account is on the blacklist;

This PR has been tested by:

Unit Tests
Manual Testing
Follow up

Extra details

### CODE REVIEW

File: pallets/staking/src/lib.rs [hunk 1]
```diff
@@ -1864,6 +1868,27 @@ pub mod pallet {
...
+            <BlackList<T>>::insert(&account_id, era);
+            Ok(())
+        }
+
+        #[pallet::weight(T::WeightInfo::remove_account_from_blacklist())]
+        pub fn remove_account_from_blacklist(
+            origin: OriginFor<T>,
+            account_id: T::AccountId,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
```
Comment (jerry-yu) [inline pallets/staking/src/lib.rs:1887]: Here maybe use privilege user is better. see  pallet_user_privelleges

**Comment by jerry-yu** [comment]
For passing CI，should run cargo fmt --all

File: pallets/staking/src/weights.rs [hunk 2]
```diff
@@ -77,6 +77,9 @@ pub trait WeightInfo {
     fn reap_stash(s: u32) -> Weight;
     fn new_era(v: u32, d: u32) -> Weight;
     fn npow_mint() -> Weight;
+    fn add_account_to_blacklist() -> Weight;
```
Comment (jerry-yu) [inline pallets/staking/src/weights.rs:80]: The funciton of weight.rs  is auto created by benchmarking.rs .  see benchmarking.rs  for details

**Review by jerry-yu — APPROVED** [review]

**Review by jerry-yu — COMMENTED** [review]
The blacklist not exactly freeze   paying reward. 
see line 2707, T::Currency::deposit_creating, that acturally func for mint balance for user be

**Review by jerry-yu — CHANGES_REQUESTED** [review]
should judge before T::Currency::deposit_creating L2707

**Comment by geb789** [comment]
@jerry-yu thanks for your comment.

**Review by jerry-yu — APPROVED** [review]

**Pull request merged**